### PR TITLE
DOC:linalg: Remove references to removed pinv2 function

### PR DESCRIPTION
**Pull request by ilayn**
Clean up remaining references to `scipy.linalg.pinv2`

Related to #16245

### CODE REVIEW

File: doc/source/tutorial/linalg.rst [hunk 1]
```diff
@@ -402,7 +400,7 @@ Generalized inverse
 ^^^^^^^^^^^^^^^^^^^
 
 The generalized inverse is calculated using the command
-:obj:`linalg.pinv` or :obj:`linalg.pinv2`. These two commands differ
+:obj:`linalg.pinv`. These two commands differ
 in how they compute the generalized inverse. The first uses the
 linalg.lstsq algorithm, while the second uses singular value
 decomposition. Let :math:`\mathbf{A}` be an :math:`M\times N` matrix,
```
Comment (j-bowhay) [inline doc/source/tutorial/linalg.rst:406]: This still contains an explanation for `pinv2`

**Comment by h-vetinari** [comment]
Not just deprecated, but removed already (I know you know that, but just to signify to other readers/reviewers that it would be nice to still get this into 1.9.0).

File: doc/source/tutorial/linalg.rst (hunk 1 above)
Comment (ilayn) [reply doc/source/tutorial/linalg.rst:406]: Ah apparently I forgot to read.

**Comment by ilayn** [comment]
True, fixed the title accordingly.

**Review by tupui — APPROVED** [review]
LGTM, thank you all.

**Comment by tupui** [comment]
Wait actually not yet `/home/circleci/repo/doc/source/tutorial/linalg.rst:354: WARNING: py:obj reference target not found: linalg.pinv`will find
`

File: doc/source/tutorial/linalg.rst [hunk 2]
```diff
@@ -353,10 +353,8 @@ where :math:`\mathbf{A}^{\dagger}` is called the pseudo-inverse of
 
 The command :obj:`linalg.lstsq` will solve the linear least-squares
 problem for :math:`\mathbf{c}` given :math:`\mathbf{A}` and
-:math:`\mathbf{y}` . In addition, :obj:`linalg.pinv` or
-:obj:`linalg.pinv2` (uses a different method based on singular value
-decomposition) will find :math:`\mathbf{A}^{\dagger}` given
-:math:`\mathbf{A}.`
+:math:`\mathbf{y}` . In addition, :obj:`linalg.pinv`will find
```
Comment (tupui) [inline doc/source/tutorial/linalg.rst:356]: I think this is due to the space
```suggestion
:math:`\mathbf{y}` . In addition, :obj:`linalg.pinv` will find
```

Comment (ilayn) [reply doc/source/tutorial/linalg.rst]: Amazing how much I can screw up just by touching a file 🤦

Comment (tupui) [reply doc/source/tutorial/linalg.rst]: It's Monday it's normal 😉

**Comment by tupui** [comment]
Yay doc CI is green 😃

**Pull request merged**